src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -233,7 +246,7 @@ void create_actor(SchedulingAlgorithmState *algorithm_state,
   entry->actor_id = actor_id;
   entry->task_counter = 0;
   /* Initialize the doubly-linked list to NULL. */
-  entry->task_queue = NULL;
+  entry->task_queue = new std::list<TaskQueueEntry *>();
```

Comment:
See comments above. Perhaps the task queue would be better encapsulated in a class?